src/Compilers/Core/Portable/CommandLine/ErrorLogger.cs

```diff
@@ -45,91 +48,102 @@ public ErrorLogger(Stream stream, string toolName, string toolFileVersion, Versi
 
         private void WriteToolInfo(string name, string fileVersion, Version assemblyVersion)
         {
-            _writer.WriteObjectStart("toolInfo");
+            _writer.WriteObjectStart("tool");
             _writer.Write("name", name);
             _writer.Write("version", assemblyVersion.ToString(fieldCount: 3));
             _writer.Write("fileVersion", fileVersion);
             _writer.WriteObjectEnd();
         }
 
-        internal void LogDiagnostic(Diagnostic diagnostic, CultureInfo culture)
+        public void LogDiagnostic(Diagnostic diagnostic)
```

Comment:
Hmm even this will break the reflection based tools.. probably it is okay to break them for once and have a cleaner API surface.